libjcat/jcat-self-test.c

```diff
@@ -1679,151 +1679,465 @@ jcat_rfc6962_func2(void)
 	g_assert_cmpint(proof_right->len, ==, 5);
 }
 
-#if 0
-inclusionProofTestVector struct {
-	     gint64 leaf;
-	 gint64 snapshot;
-	    GPtrArray *proof;
+struct inclusionProofTestVector {
+	gint64 leaf;
+	gint64 snapshot;
+	GPtrArray *proof;
+};
+
+static struct inclusionProofTestVector *
+inclusionProofTestVectorNew(void)
+{
+	struct inclusionProofTestVector *iptv = g_new0(struct inclusionProofTestVector, 1);
+	iptv->proof = g_ptr_array_new_with_free_func((GDestroyNotify)g_bytes_unref);
+	return iptv;
+}
+
+static void
+inclusionProofTestVectorFree(struct inclusionProofTestVector *iptv)
+{
+	g_ptr_array_unref(iptv->proof);
+}
+
+static char
+decode_nibble(char h)
+{
+	if (h >= '0' && h <= '9')
+		return h - '0';
+	if (h >= 'a' && h <= 'f')
+		return h - 'a' + 10;
+	if (h >= 'A' && h <= 'F')
+		return h - 'A' + 10;
+	return 0;
+}
+
+static GBytes *
+dh(const char *s, int expected_size)
+{
+	char buf[32];
```

Comment:
gchar please, and also it's a good idea to explicitly zero-set the array to avoid confusion later, e.g. `gchar buf[32] = {0};`